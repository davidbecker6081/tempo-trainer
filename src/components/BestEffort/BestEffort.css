.best-effort {
  justify-content: space-around;
  margin-bottom: 1rem;
  padding: 1em 0;
  width: 100%;
  background-color: #252525;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 2px #25252563;
  color: #FFFFFF;
  font-size: 1.2em;
}

.best-effort p {
  margin: 0;
}

.best-effort-title,
.best-effort-timerange {
  flex-direction: column;
  width: 45%;
}

.best-effort-title {
  font-size: 1.3em;
}

.best-effort-average {
  font-size: 1.3em;
  font-weight: 700;
}

@media screen and (max-width: 770px) {
  .best-effort {
    width: 45%;
    margin-right: 10px;
  }
}
