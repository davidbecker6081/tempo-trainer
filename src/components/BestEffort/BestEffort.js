import React from 'react';
import PropTypes from 'prop-types';
import './BestEffort.css';

const BestEffort = ({
  channelSet,
  timeForCalc,
  effort,
  timeRange,
}) => {
  const effortCleaner = {
    power: Math.round(effort),
    heartRate: Math.round(effort),
    cadence: Math.round(effort),
    temperature: Math.round(effort * 10) / 10,
    elevation: Math.round(effort),
    speed: Math.round(effort * 100) / 100,
  };

  const timeCleaner = {
    low: Math.round((timeRange.low / 60000) * 10) / 10,
    high: Math.round((timeRange.high / 60000) * 10) / 10,
  };

  return (
    <article>
      <p>Best <span>{timeForCalc}</span> Minute</p>
      <p>{channelSet.toUpperCase()}</p>
      <p>Average</p>
      <p>{effortCleaner[channelSet]}</p>
      <p>Minute {timeCleaner.low} to Minute {timeCleaner.high}</p>
    </article>
  );
};

export default BestEffort;

BestEffort.propTypes = {
  channelSet: PropTypes.string.isRequired,
  timeForCalc: PropTypes.number.isRequired,
  effort: PropTypes.number.isRequired,
  timeRange: PropTypes.shape({
    low: PropTypes.number,
    high: PropTypes.number,
  }).isRequired,
};
