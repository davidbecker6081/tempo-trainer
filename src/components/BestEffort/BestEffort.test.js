import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import BestEffort from './BestEffort';

describe('BestEffort', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      channelSet: '',
      timeForCalc: 0,
      effort: 0,
      timeRange: {
        low: 0,
        high: 0,
      },
    };
  });

  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('timeForCalc is defined', () => {
    beforeEach(() => { mockProps.timeForCalc = 1; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('effort is defined', () => {
    beforeEach(() => { mockProps.effort = 1; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('timeRange is defined', () => {
    beforeEach(() => {
      mockProps.timeRange.low = 0;
      mockProps.timeRange.high = 20;
    });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
