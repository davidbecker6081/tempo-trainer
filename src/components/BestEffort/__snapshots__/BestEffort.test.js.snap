// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestEffort channelSet is defined should render correclty 1`] = `undefined`;

exports[`BestEffort effort is defined should render correclty 1`] = `undefined`;

exports[`BestEffort timeForCalc is defined should render correclty 1`] = `undefined`;

exports[`BestEffort timeRange is defined should render correclty 1`] = `undefined`;
